### PR TITLE
[OUR415-314] Improves our test story with test templates and helpers for search; adds `BrowserRefinementList` component test

### PR DESCRIPTION
**Pull request by rosschapman**
I wrote our first component test of a component with Algolia hooks. This code establishes a precedent but as more tests are written we might come up with improvements to the "test harness" to increase ease of writing tests.

### CODE REVIEW

File: app/components/search/Refinements/BrowseRefinementList.test.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import { InstantSearch } from "react-instantsearch-core";
+import { render, screen, waitFor } from "@testing-library/react";
+import BrowseRefinementList from "components/search/Refinements/BrowseRefinementList";
+import { createSearchClient } from "../../../../test/helpers/createSearchClient";
+
+describe("BrowseRefinementList", () => {
+  test("renders the correct the number of categories", async () => {
```
Comment (adriencyberspace) [inline app/components/search/Refinements/BrowseRefinementList.test.tsx:8]: **question (non-blocking):** There are 12 categories in searchClient but expected is 10. Is there anything in the repo that explains this?

File: app/components/search/Refinements/BrowseRefinementList.tsx [hunk 2]
```diff
@@ -39,10 +39,12 @@ const BrowseRefinementList = ({ attribute, transform }: Props) => {
     setChecked(checked);
   };
 
+  // console.log(items);
```
Comment (adriencyberspace) [inline app/components/search/Refinements/BrowseRefinementList.tsx:42]: **chore:** Let's get rid of this

File: test/helpers/createSearchClient.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+ *   </InstantSearch>
+ * );
+ * ```
+ *
+ * @param options Additional customizations of the search response
+ * @returns
+ */
+export function createSearchClient(options: Options) {
+  return {
+    search: (requests: any) =>
```
Comment (adriencyberspace) [inline test/helpers/createSearchClient.ts:35]: **question (if-minor):** In the interest of minimizing lint warnings, can we change these 2 'any' types in this file or can we lint ignore this file?

File: test/templates/Template.test.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+/**
+ * Template for a simple component test
```
Comment (adriencyberspace) [inline test/templates/Template.test.tsx:2]: **praise:** Good idea

**Review by adriencyberspace — APPROVED** [review]
Please answer / address my comments but looks good to me otherwise!

File: app/components/search/Refinements/BrowseRefinementList.test.tsx (hunk 1 above)
Comment (adriencyberspace) [inline app/components/search/Refinements/BrowseRefinementList.test.tsx:8]: **nitpick (non-blocking):** Could you remove the second "the"?

File: test/helpers/createSearchClient.ts (hunk 3 above)
Comment (rosschapman) [reply test/helpers/createSearchClient.ts:35]: Oops yeah, forgot to add ignore statements. I was going to clean up all the warnings in a separate PR so I can do it there.

File: app/components/search/Refinements/BrowseRefinementList.test.tsx (hunk 1 above)
Comment (rosschapman) [reply app/components/search/Refinements/BrowseRefinementList.test.tsx]: Good question. I updated the test name to indicate more explicitly what this is testing.

**Pull request merged**